From 361f6c06b2fc0db7d935650735043d49f358b7ff Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Tue, 15 Sep 2015 14:27:00 +0800
Subject: [PATCH 0482/5242] MLK-11561-2 regulator: anatop: set default voltage
 selector for pcie

commit  818b3b9001809377494ace2711a91e13dfc07267 from
https://source.codeaurora.org/external/imx/linux-imx.git

The code reads the default voltage selector from its register.
If the default voltage selector is 0 which results in faulty
behaviour of this regulator driver.

This patch sets a default voltage selector for vddpcie-phy if
it is not set in the register.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/anatop-regulator.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/anatop-regulator.c b/drivers/regulator/anatop-regulator.c
index c2e168c..23dfcd9 100644
--- a/drivers/regulator/anatop-regulator.c
+++ b/drivers/regulator/anatop-regulator.c
@@ -344,7 +344,7 @@ static int anatop_regulator_probe(struct platform_device *pdev)
 			sreg->sel = 22;
 
 		/* set the default voltage of the pcie phy to be 1.100v */
-		if (!sreg->sel && !strcmp(rdesc->name, "vddpcie"))
+		if (!sreg->sel && !strcmp(rdesc->name, "vddpcie-phy"))
 			sreg->sel = 0x10;
 
 		if (!sreg->bypass && !sreg->sel) {
-- 
1.7.9.5

